pandas/tests/indexes/datetimes/test_partial_slicing.py

```diff
@@ -236,7 +236,7 @@ def test_partial_slice_second_precision(self):
         rng = date_range(
             start=datetime(2005, 1, 1, 0, 0, 59, microsecond=999990),
             periods=20,
-            freq="US",
+            freq="us",
```

Comment:
does this PR make `to_offset('US')` suddently start failing? if so, I don't think we can do this yet (at least, not before https://github.com/pandas-dev/pandas/pull/56346/files ?)

Reply:
> does this PR make `to_offset('US')` suddently start failing? if so, I don't think we can do this yet (at least, not before https://github.com/pandas-dev/pandas/pull/56346/files ?)

yes, this test didn't fail for uppercase frequency `"US"` before this PR.

Reply:
a small correction to the comment above: 
we fixed the `test_partial_slice_second_precision` in PR  DEPR: lowercase strings denoting freq for Week, MonthBegin, MonthEnd, etc. #56346 
but so far on main we have `to_offset('US')` in this test and it pass.